app/views/u2f_registrations/index.html.slim

```diff
@@ -0,0 +1,34 @@
+- content_for(:navbar_content)
+
+.register-2fa
+  h1 Welcome to 2fa registration
+
+  form.js-register-u2f method="post" action=u2f_registrations_path
+    = hidden_field_tag :authenticity_token, form_authenticity_token
+    input type="hidden" name="u2f_app_id" value=(@app_id)
+    input type="hidden" name="u2f_registration_requests" value=(@registration_requests.as_json.to_json)
+    input type="hidden" name="u2f_registered_keys" value=(@registered_keys.as_json.to_json)
+    input type="hidden" name="u2f_response"
+    div.form-group
+      input.form-control type="text" name="name" placeholder="Enter a nickname for this security key"
+    button.btn.btn-primary type="submit" Register a new device
+
+  - if @u2f_registrations.any?
+    hr
+
+    h2 U2F Keys
+
+    table.table
+      thead
+        tr
+          th scope="col" Name
+          th scope="col" Registered
+          th scope="col" Last Used
+          th
+      tbody
+        - @u2f_registrations.each do |u2f|
+          tr
+            th scope="row" = u2f.name || "Anonymous Key"
```

Comment:
Your JavaScript is showing ;-) empty strings are truthy in Ruby, this should probably be

```
(u2f.name.blank? ? u2f.name : "Anonymous Key")
```

Reply:
Good catch. I had coded (and tested) the default before I added the `name` field to the registration form - so I had tested it only against `nil` values 🤦‍♂️